apps/prairielearn/assets/scripts/instructorAssessmentInstancesClient.ts

```diff
@@ -0,0 +1,512 @@
+        .popover({
+          sanitize: false,
+          placement: 'right',
+          title() {
+            const row = $(this).data('row');
+            return row.action === 'set_time_limit_all'
+              ? 'Change Time Limits'
+              : row.open
+                ? 'Change Time Limit'
+                : 'Re-Open Instance';
+          },
+          container: 'body',
+          html: true,
+          trigger: 'click',
+          content: window.timeLimitEditPopoverContent,
+        })
+        .on('show.bs.popover', function () {
+          $($(this).data('bs.popover').getTipElement()).css('max-width', '350px');
+          $(this).find('.select-time-limit').change();
+        });
+    },
+  });
+
+  $(document).on('keydown', (event) => {
+    if (
+      (event.ctrlKey || event.metaKey) &&
+      String.fromCharCode(event.which).toLowerCase() === 'f'
+    ) {
+      $('.fixed-table-toolbar .search input').focus();
+      event.preventDefault();
+    }
+  });
+
+  $('#deleteAssessmentInstanceModal').on('show.bs.modal', function (event: any) {
+    const button = $(event.relatedTarget); // Button that triggered the modal
+    const uid = button.data('uid'); // Extract info from data-* attributes
+    const name = button.data('name');
+    const number = button.data('number');
+    const group_name = button.data('group-name');
+    const uid_list = button.data('uid-list');
+    const date_formatted = button.data('date-formatted');
+    const score_perc = button.data('score-perc');
+    const assessment_instance_id = button.data('assessment-instance-id');
+    const modal = $(this);
+
+    modal.find('form').submit((e) => {
+      e.preventDefault();
+      $.post(
+        $(e.target).attr('action') ?? '',
+        $(e.target).serialize(),
+        function () {
+          window.refreshTable();
+        },
+        'json',
+      );
+      modal.modal('hide');
+    });
+
+    modal.find('.modal-uid').text(uid);
+    modal.find('.modal-name').text(name);
+    modal.find('.modal-group-name').text(group_name);
+    modal.find('.modal-uid-list').text(uid_list);
+    modal.find('.modal-number').text(number);
+    modal.find('.modal-date').text(date_formatted);
+    modal.find('.modal-score-perc').text(score_perc);
+    modal.find('.modal-assessment-instance-id').val(assessment_instance_id);
+  });
+
+  $('#deleteAllAssessmentInstancesModal').on('show.bs.modal', function () {
+    const modal = $(this);
+
+    modal.find('form').submit((e) => {
+      e.preventDefault();
+      $.post(
+        $(e.target).attr('action') ?? '',
+        $(e.target).serialize(),
+        function () {
+          window.refreshTable();
+        },
+        'json',
+      );
+      modal.modal('hide');
+    });
+  });
+
+  $('[data-toggle="modal"]').on('click', function (e) {
+    e.stopPropagation(); // Keep click from changing sort
+    $($(e.currentTarget).data('target')).modal('show');
+  });
+});
+
+window.refreshTable = function () {
```

Comment:
I don't think any of these need to be window globals, they can be local functions.
```suggestion
function refreshTable() {
```

Same with the other functions. This can help you get rid of the typing at the top, and should make the calls simpler.

Reply:
`bootstrap-table` is looking for these functions on the global object. If I try to declare these as local functions I get results that looks like the following: 
![Screenshot 2024-04-11 at 1 18 47 PM](https://github.com/PrairieLearn/PrairieLearn/assets/114321556/3f95fda6-c06c-40e3-84d2-fe7113b4a643)

Reply:
While that's true for at least some of these, `refreshTable` in particular does not need to be. Spot checking, `updateTotals` doesn't need to be either. Only functions which are referenced by name in `data-*` attributes or in `onclick="..."` and similar handlers need to be attached to `window`. All other functions can be local to this module.

Reply:
You can also move these attributes to the call of bootstrapTable, this way you don't need them to be in the global context. See: https://github.com/PrairieLearn/PrairieLearn/blob/0a43a655fcc2ba5109244a220062efe0278f4688/apps/prairielearn/src/pages/instructorAssessmentManualGrading/assessmentQuestion/assessmentQuestion.ejs#L93-L183

Reply:
Thanks! I moved these into the `columns` option of the `bootstrapTable` in [18023da](https://github.com/PrairieLearn/PrairieLearn/pull/9725/commits/18023da877be9d5cd204151d7255b3c25b0778af) similar to above. I was able to remove almost all of the functions from the global object (additional cleanup in [47c6453](https://github.com/PrairieLearn/PrairieLearn/pull/9725/commits/47c64535f56ac03c1e7f83274b8604ac1cd6330f)) with the exception of `popoverSubmitViaAjax`.